matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/timeline/TimelineSettings.kt

```diff
@@ -27,5 +27,14 @@ data class TimelineSettings(
         /**
          * If true, will build read receipts for each event.
          */
-        val buildReadReceipts: Boolean = true
-)
+        val buildReadReceipts: Boolean = true,
+        /**
+         * The root thread eventId if this is a thread timeline, or null if this is NOT a thread timeline
+         */
+        val rootThreadEventId: String? = null) {
+
+    /**
+     * Returns true if this is a thread timeline or false otherwise
+     */
+    fun isThreadTimeline() = rootThreadEventId != null
```

Comment:
Why do you think is a good idea to add more properties in a data class and not simply parse the existing `rootThreadEventId` when we need it. This function is to avoid repeating this check only when is needed

Reply:
Adding a `val` in the body of a `data class` is not the same thing than adding a new property (i.e. a `val` in the `constructor`) of a data class.
Replacing this line by `val isThreadTimeline = rootThreadEventId != null` will just compute this val once in the constructor.
Anyway this is a minor remark!

Reply:
Actually it's interesting to see another perspective! So this is a battle between `CPU` vs `RAM`
- `fun isThreadTimeline() = rootThreadEventId != null` will compute it every time but will never store anything else in the ram
-  `val isThreadTimeline = rootThreadEventId != null` will compute it only once and use more ram to store it

Both approaches work for me, yes its minor but interesting :)